### ISSUE TITLE
Clang-format

### ISSUE THREAD

**Issue by Rombur** [ORNL-CEES/DataTransferKit#204]
Should we stay with clang-format 3.9 or move to clang-format 4.0?

**Comment by dalg24** [comment]
Did you check if this would introduce much changes?

**Comment by Rombur** [comment]
Nope, it's more of a general question about our policy. Do we want to stick with clang-format 3.9 or use the latest version?

**Comment by aprokop** [comment]
Is there a compelling reason? Personally, I don't care as long as spack supports whatever version we want to use.

**Comment by Rombur** [comment]
Difference between clang-format 4.0 and clang-format 3.9 on #205 

```-- Configuring done
-- Generating done
-- Build files have been written to: /scratch/source/trilinos/release/DataTransferKit/build
--- a/scratch/source/trilinos/release/DataTransferKit/packages/Kokkos/test/ViewTestCudaHelpers_def.hpp
+++ b/scratch/source/trilinos/release/DataTransferKit/packages/Kokkos/test/ViewTestCudaHelpers_def.hpp
@@ -117,13 +117,13 @@
     }
     else if ( 2 == num_dims )
     {
-        fill2dKernel<Scalar><<<num_blocks, block_size>>>( dtk_view.data(),
-                                                          dims[0], dims[1] );
+        fill2dKernel<Scalar>
+            <<<num_blocks, block_size>>>( dtk_view.data(), dims[0], dims[1] );
     }
     else if ( 3 == num_dims )
     {
-        fill3dKernel<Scalar><<<num_blocks, block_size>>>( dtk_view, dims[0],
-                                                          dims[1], dims[2] );
+        fill3dKernel<Scalar>
+            <<<num_blocks, block_size>>>( dtk_view, dims[0], dims[1], dims[2] );
     }
 
     // NOTE: Synchronization is required because kernels are launched

1 file(s) not formatted properly:
    /scratch/source/trilinos/release/DataTransferKit/packages/Kokkos/test/ViewTestCudaHelpers_def.hpp
```

**Comment by dalg24** [comment]
That's all?

**Comment by Rombur** [comment]
yep

**Comment by aprokop** [comment]
Hmm, I was hoping that once Fedora 26 is released (~ June) it would have llvm-3.9 in it, and I won't need to use an extra. But don't consider this as a showstopper, go with whatever you decide.

**Comment by aprokop** [comment]
I guess we are all OK with moving to clang-4. Closing this as it is inherited by #208.
